src/plugins/dashboard/common/types.ts

```diff
@@ -85,3 +86,40 @@ export type SavedDashboardPanel730ToLatest = Pick<
   readonly id?: string;
   readonly type: string;
 };
+
+export interface DashboardListSource {
+  name: string;
+  listProviderFn: () => Observable<DashboardListItem>;
+}
+
+export type DashboardListSources = DashboardListSource[];
+
+export interface DashboardListItem {
+  id: string;
+  title: string;
+  type: string;
+  description: string;
+  url: string;
+  listType: string;
+}
+
+export type DashboardListItems = DashboardListItem[];
+
+export type DashboardCreators = DashboardCreator[];
+
+export type DashboardCreatorFn = (event: any, history: any) => void;
+export type DashboardItemCreatorClickHandler = (
+  creatorFn: DashboardCreatorFn
+) => (event: any) => void;
+
+export interface DashboardCreator {
+  id: string; // key identifier for creator plugin/module
+  name: string; // display name for create link
+  i18nId?: string; // unique identifier for react-intl FormattedMessage
+  i18nDescription?: string; // helpful description of this i18n context
+  i18nDefaultMessage?: string; // FormattedMessage default message, in tag-replace format
+  i18nDefaultValues?: object; // FormattedMessage default value
+  i18nEntityName?: string; // translation key
+  defaultText: string; // default translation text
```

Comment:
Why are we providing such granular information about i18n keys and values? can we just pass the locallized string? I'll admit i'm not familiar with how localization is setup in OSD today so if there is an inherent reason we cant do what i think we can, id love to know :)

Reply:
Yes, correct.  Simplified in upcoming commit.